File: app/src/main/java/com/example/smishingdetectionapp/chat/ChatMessage.java
package com.example.smishingdetectionapp.chat;

/**
 * Represents a chat message in the conversation between user and bot.
 */
public class ChatMessage {

    // Constants to identify message sender type
    public static final int USER = 0;
    public static final int BOT = 1;

    // The content of the message
    private String message;

    // Type of message (USER or BOT)
    private int messageType;

    /**
     * Constructs a ChatMessage with specified content and sender type.
     *
     * @param message The message content.
     * @param messageType The sender type (USER or BOT).
     */
    public ChatMessage(String message, int messageType) {
        this.message = message;
        this.messageType = messageType;
    }

    /**
     * Gets the message content.
     *
     * @return The message text.
     */
    public String getMessage() {
        return message;
    }

    /**
     * Gets the message type.
     *
     * @return 0 if sent by USER, 1 if sent by BOT.
     */
    public int getMessageType() {
        return messageType;
    }
}

File: shared/src/commonMain/kotlin/com/example/smishingdetectionapp/data/model/chat/ChatMessage.kt
package com.example.smishingdetectionapp.data.model.chat

import kotlinx.serialization.Serializable
import com.example.smishingdetectionapp.util.currentTimeMillis

@Serializable
data class ChatMessage(
    val id: Long = 0,
    val sender: String,
    val text: String,
    val timestamp: Long
) {
    companion object {
        const val USER = "USER"
        const val BOT = "BOT"

        fun userMessage(text: String, timestamp: Long = currentTimeMillis()): ChatMessage {
            return ChatMessage(sender = USER, text = text, timestamp = timestamp)
        }

        fun botMessage(text: String, timestamp: Long = currentTimeMillis()): ChatMessage {
            return ChatMessage(sender = BOT, text = text, timestamp = timestamp)
        }
    }

    val isFromUser: Boolean
        get() = sender == USER
}